src/components/Button/index.tsx

```diff
@@ -3,38 +3,81 @@ import {
   Pressable,
   PressableProps,
   ViewStyle,
-  TextStyle,
   StyleProp,
-  Text,
-  View,
   GestureResponderEvent,
 } from 'react-native';
 import {useStyles} from './styles';
+import {useTheme, ThemeType} from '../Theme';
 import {useFormContext} from '../Form';
+import Text from '../Text';
+import Icon, {IIconProps} from '../Icon';
+
+export type TSize = 'mini' | 'small' | 'medium' | 'large';
 
 export interface IButtonProps extends PressableProps {
-  size?: 'mini' | 'small' | 'medium' | 'large';
-  shape?: 'flat' | 'round' | 'circle';
-  variant?: 'contained' | 'outlined';
-  color?: 'primary' | 'secondary';
+  label?: string;
+  size?: TSize;
+  shape?: 'curved' | 'round' | 'flat';
+  variant?: 'contained' | 'outlined' | 'empty';
+  error?: boolean;
   type?: 'submit' | 'reset';
   disabled?: boolean;
-  iconLeft?: React.ReactNode;
-  iconRight?: React.ReactNode;
+  icon?: IIconProps['name'];
+  iconPosition?: 'left' | 'right';
   style?: StyleProp<ViewStyle>;
-  textStyle?: StyleProp<TextStyle>;
 }
 
+const getTextColor = ({
+  theme,
+  variant,
+  disabled,
+  error,
+  pressed,
+}: IButtonProps & {theme: ThemeType; pressed?: boolean}) => {
+  if (disabled) {
+    return theme.text.disabled;
+  }
+  if (variant === 'contained') {
+    return theme.primary.contrast2;
+  }
+  if (variant === 'outlined') {
+    if (pressed) {
+      if (error) {
+        return theme.error.dark;
+      }
+
+      return theme.primary.dark;
+    }
+    if (error) {
+      return theme.error.main;
+    }
+
+    return theme.primary.main;
+  }
+  if (variant === 'empty') {
+    if (error) {
+      return theme.error.main;
+    }
+    return theme.primary.contrast;
+  }
+};
```

Comment:
I think that it's better to put such styles related things to `styles.ts`. What do you think?

Reply:
I also think the same, however in this case it is a prop rather than a style

Reply:
I might add it to styles as an object and import it tho. What d u thin?